File: modulo_2_must_go/fast_odd_even.java
import java.time.*;
import java.time.temporal.ChronoUnit;
import java.util.*;

class fast_odd_even {

  public static void andMethod(final int ITS) {
    int temp = 0;

    for (int i = 0; i < ITS; i++) {
      if ((i & 1) == 0) {
        temp++;
      }
    }
    System.out.printf("%d\n", temp);
  }

  public static void moduloMethod(final int ITS) {
    int temp = 0;

    for (int i = 0; i < ITS; i++) {
      if (i % 2 == 0) {
        temp++;
      }
    }
    System.out.printf("%d\n", temp);
  }

  private static double Average(List<Long> l) {
    Long sum = 0L;
    if (!l.isEmpty()) {
      for (Long e : l) {
        sum += e;
      }
      return sum.doubleValue() / l.size();
    }
    return sum;
  }

  public static void main(String[] args) {
    List<Long> aC = new ArrayList<>();
    List<Long> bC = new ArrayList<>();
    final int ITS = 10000000;

    Long start;
    Long stop;

    for (int i = 0; i < 100; i++) {
      start = ChronoUnit.MICROS.between(Instant.EPOCH, Instant.now());
      andMethod(ITS);
      stop = ChronoUnit.MICROS.between(Instant.EPOCH, Instant.now());
      aC.add(stop - start);

      start = ChronoUnit.MICROS.between(Instant.EPOCH, Instant.now());
      andMethod(ITS);
      stop = ChronoUnit.MICROS.between(Instant.EPOCH, Instant.now());
      bC.add(stop - start);
    }

    double aAvg = Average(aC);
    double bAvg = Average(bC);

    System.out.printf("and    avg: %.2f uSec\n", aAvg);
    System.out.printf("modulo avg: %.2f uSec\n", bAvg);
  }
}

File: modulo_2_must_go/fast_odd_even.py
#!/usr/bin/env python3

import time


def moduloMethod(ITS):
    temp = 0
    for i in range(0, ITS):
        if i % 2 == 0:
            temp += 1
    print(temp)


def andMethod(ITS):
    temp = 0
    for i in range(0, ITS):
        if not i & 2:
            temp += 1
    print(temp)


def Average(lst):
    return sum(lst) / len(lst)


aC = []
bC = []

ITS = 10000000

for i in range(0, 100):
    start = time.time_ns()
    andMethod(ITS)
    stop = time.time_ns()
    aC.append(stop - start)

    start = time.time_ns()
    moduloMethod(ITS)
    stop = time.time_ns()
    bC.append(stop - start)

aAvg = Average(aC) / 1000

bAvg = Average(bC) / 1000

print("and    method average %d uSec" % (aAvg))
print("modulo method average %d uSec" % (bAvg))
